src/averageAges.js

```diff
@@ -55,7 +92,34 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter((p) => p.mother !== null);
+  const sons = children.filter((p) => p.sex === 'm');
+
+  const childrenArr = onlyWithSon ? sons : children;
+
+  const childrenRes = childrenArr.reduce((acc, child) => {
+    const mother = people.find((p) => p.name === child.mother);
+    const diff = mother ? child.born - mother.born : 0;
+
+    return mother
+      ? [
+        ...acc,
+        {
+          ...child,
+          ageMotherDiff: diff,
+        },
+      ]
+      : acc;
+  }, []);
```

Comment:
Looks very complicated. This task can be solved much easier. Try to find another solution